elasticsearch/bulk/bulk.go

```diff
@@ -327,3 +349,43 @@ func (b *Bulk) getIndexName(collectionName, actionIndexName string) string {
 
 	return indexName
 }
+
+func (b *Bulk) executeSinkResponseHandler(batchActions []*document.ESActionDocument, errorData map[string]string) {
+	if b.sinkResponseHandler == nil {
+		return
+	}
+
+	for _, action := range batchActions {
+		key := getActionKey(*action)
```

Comment:
how can we avoid creating same action key twice(one for real action one for response handler)

Reply:
When i looked as following part, we avoid to create same action 🤔 

```
key := getActionKey(action)
if batchIndex, ok := b.batchKeys[key]; ok {
	b.batch[batchIndex] = BatchItem{
		Action: &actions[i],
		Bytes:  value,
	}
} else {
	b.batch = append(b.batch, BatchItem{
		Action: &actions[i],
		Bytes:  value,
	})
	b.batchKeys[key] = b.batchIndex
	b.batchIndex++
}
```

We are calling `b.executeSinkResponseHandler(getActions(batchItems), errorData)` only given batchItems of chunk.